lib/src/Base/Func/EnumerateFunctionImplementation.cxx

```diff
@@ -148,5 +149,16 @@ void EnumerateFunctionImplementation::load(Advocate & adv)
     upperBound_ = Indices(getDimension(), std::numeric_limits<UnsignedInteger>::max());
 }
 
+/* Returns the marginal enumerate function */
+EnumerateFunction EnumerateFunctionImplementation::getMarginal(const Indices &) const
+{
+  throw NotYetImplementedException(HERE) << "In EnumerateFunctionImplementation::getMarginal";
+}
+
+/* Returns the marginal enumerate function */
+EnumerateFunction EnumerateFunctionImplementation::getMarginal(const UnsignedInteger) const
+{
+  throw NotYetImplementedException(HERE) << "In EnumerateFunctionImplementation::getMarginal";
```

Comment:
it should call getMarginal(Indices) so that we dont have to override it for each class

Reply:
Fixed at d17f1462144022b009756e26e0d7de62313a3207

Reply:
Oups: it fails.